com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/app/discovery/ZigBeeNetworkDiscoverer.java

```diff
@@ -459,21 +459,9 @@ private boolean getAssociatedNodes(final int networkAddress) throws InterruptedE
      */
     private void addNode(final IeeeAddress ieeeAddress, int networkAddress) {
         logger.debug("{}: NWK Discovery add node {}", ieeeAddress, networkAddress);
-        ZigBeeNode node = networkManager.getNode(ieeeAddress);
-        if (node != null) {
-            if (node.getNetworkAddress() != networkAddress) {
-                logger.debug("{}: NWK Discovery updated network address to {}", ieeeAddress, networkAddress);
-            }
-            node.setNodeState(ZigBeeNodeState.ONLINE);
-            node.setNetworkAddress(networkAddress);
-            networkManager.updateNode(node);
-            return;
-        }
-
-        node = new ZigBeeNode(networkManager, ieeeAddress);
+        ZigBeeNode node = new ZigBeeNode(networkManager, ieeeAddress);
+        node.setNodeState(ZigBeeNodeState.ONLINE);
         node.setNetworkAddress(networkAddress);
```

Comment:
I wonder, while you are changing the API anyway, if its worth to make this `networkAddress` a required parameter for the constructor. Some places in the code already rely on it and fail with a NPW if the `networkAddress` is not set on the node. WDYT?

Reply:
I guess we could, but we don't always know the NWK address when the node is instantiated. What is REQUIRED is the EUI. I've not checked, but I suspect if we force the NWK to be provided in the constructor that it will make other areas more complex where NWK is not known.

I'll have a look at this though - I guess we could add an extra constructor at least.

Reply:
I've added a new constructor that also takes the NWK address, and have changed applicable code to use this.

Reply:
Thanks